third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js

```diff
@@ -278,39 +278,36 @@ UserPromptingChecker.prototype.promptUserForUnknownApp_ = function(
  */
 UserPromptingChecker.prototype.runPromptDialog_ = function(
     clientAppId, userPromptDialogData, promiseResolver) {
-  const dialogPromise = GSC.PopupWindow.Server.runModalDialog(
-      USER_PROMPT_DIALOG_URL, USER_PROMPT_DIALOG_WINDOW_OPTIONS_OVERRIDES,
-      userPromptDialogData);
-  dialogPromise.then(
-      function(dialogResult) {
-        if (dialogResult) {
-          goog.log.info(
-              this.logger,
-              'Granted permission to client App with id "' + clientAppId +
-                  '" based on the "grant" user selection');
-          this.storeUserSelection_(clientAppId, true);
-          promiseResolver.resolve();
-        } else {
-          goog.log.info(
-              this.logger,
-              'Rejected permission to client App with id "' + clientAppId +
-                  '" based on the "reject" user selection');
-          this.storeUserSelection_(clientAppId, false);
-          promiseResolver.reject(new Error(
-              'Reject permission based on the "reject" user selection'));
-        }
-      },
-      function() {
-        goog.log.info(
-            this.logger,
-            'Rejected permission to client App with id "' + clientAppId +
-                '" because of the user cancellation of the prompt dialog');
-        this.storeUserSelection_(clientAppId, false);
-        promiseResolver.reject(new Error(
-            'Rejected permission because of the user cancellation of the prompt ' +
-            'dialog'));
-      },
-      this);
+  
+  chrome.windows.create({url: USER_PROMPT_DIALOG_URL, width: 300, height: 180, type: "popup"});
+
+  // #toclean
+  // var dialogPromise = GSC.PopupWindow.Server.runModalDialog(
```

Comment:
Now that we know that we're going to fix PopupWindow.Server to work in extensions too, is there really a need in commenting out all of that?

Maybe we should start from a PR that would start this update of PopupWindow.Server?

Reply:
I thought I'd start with the UI migration and then the backend part, WDYT?

Reply:
Sure, but we don't have to break the code base on Github while doing that. It's a common principle in the software development: try to maintain the central repository in a good shape even in the middle of complex multi-commit changes. For example, see Abseil's "Live at Head" philosophy: https://abseil.io/about/philosophy#upgrade-support